### PR TITLE
Improvements to the mobile navbar

### PR DESCRIPTION
**Pull request by NeutronicMC**
Improves reachability, consistency between menus, and allows for the possibility of a customizable navbar.

CleanShot 2023-02-05 at 14 40 43@2x
CleanShot 2023-02-05 at 14 40 16@2x
CleanShot 2023-02-05 at 14 39 57@2x

### CODE REVIEW

**Comment by NeutronicMC** [comment]
After discussion, here's the current state of the PR:
https://user-images.githubusercontent.com/17936976/216884410-16d460a6-ee27-4519-8e86-26e5e11743b7.mp4

File: layouts/default.vue [hunk 1]
```diff
@@ -405,6 +345,33 @@ const auth = await useAuth()
...
+  {
+    label: 'Resource Packs',
+    href: '/resourcepacks',
+  },
+  {
+    label: 'Modpacks',
+    href: '/modpacks',
+  },
+]
+const SEARCH_PAGES = ['/mods', '/plugins', '/datapacks', '/shaders', '/resourcepacks', '/modpacks']
```
Comment (MagnusHJensen) [inline layouts/default.vue:374]: No need to duplicate routes.

File: layouts/default.vue [hunk 2]
```diff
@@ -439,9 +406,15 @@ export default defineNuxtComponent({
     // }
     return {}
   },
+  computed: {
+    isOnSearchPage() {
+      return SEARCH_PAGES.some((page) => this.$route.path.startsWith(page))
```
Comment (MagnusHJensen) [inline layouts/default.vue:411]:
```suggestion
      return NAV_ROUTES.some((route) => this.$route.path.startsWith(route.href))
```

File: layouts/default.vue (hunk 1 above)
Comment (NeutronicMC) [reply layouts/default.vue:374]: Oops, I totally missed that! I'll fix it in the morning.

**Pull request merged**